### PR TITLE
feat: add `UnboundPredicate::negate()`

### PR DESCRIPTION
**Pull request by sdd**
Adds a `negate()` method to `UnboundPredicate`, which will be required as part of #150

### CODE REVIEW

**Comment by sdd** [comment]
@liurenjie1024: this is ready for review 👍🏼

File: crates/iceberg/src/expr/mod.rs [hunk 1]
```diff
@@ -30,7 +30,7 @@ pub use predicate::*;
 /// The discriminant of this enum is used for determining the type of the operator, see
 /// [`PredicateOperator::is_unary`], [`PredicateOperator::is_binary`], [`PredicateOperator::is_set`]
 #[allow(missing_docs)]
-#[derive(Debug, Clone, Copy)]
+#[derive(Debug, Clone, Copy, PartialEq)]
```
Comment (liurenjie1024) [inline crates/iceberg/src/expr/mod.rs:33]: I think they are also `Eq`?

**Review by liurenjie1024 — APPROVED** [review]
LGTM, thanks!

**Comment by liurenjie1024** [comment]
cc @Xuanwo @ZENOTME @Fokko PTAL

**Pull request merged**